core/src/main/java/org/infinispan/functional/impl/EntryViews.java

```diff
@@ -0,0 +1,273 @@
+         // hence we cannot distinguish between create or update.
+         //notifier.notifyOnWrite(EntryViews.readOnly(entry));
+         return null;
+      }
+
+      @Override
+      public Void remove() {
+         entry.setRemoved(true);
+         entry.setChanged(true);
+         // For remove write-only listener events, create a value-less read entry view
+         //notifier.notifyOnWrite(EntryViews.noValue(entry.getKey()));
+         return null;
+      }
+   }
+
+   private static final class ReadWriteViewImpl<K, V> implements ReadWriteEntryView<K, V> {
+      final ListenerNotifier<K, V> notifier;
+      final CacheEntry<K, V> entry;
+
+      private ReadWriteViewImpl(CacheEntry<K, V> entry, ListenerNotifier<K, V> notifier) {
+         this.entry = entry;
+         this.notifier = notifier;
+      }
+
+      @Override
+      public K key() {
+         return entry.getKey();
+      }
+
+      @Override
+      public Optional<V> find() {
+         return entry == null ? Optional.empty() : Optional.ofNullable(entry.getValue());
+      }
+
+      @Override
+      public Void set(V value, MetaParam.Writable... metas) {
+         if (!entry.isCreated()) {
+            // TODO: Modify listeners
+            //notifier.notifyOnModify(EntryViews.readOnly(key, prev), EntryViews.readOnly(key, iv));
+         } else {
+            // TODO: Created listeners
+            // notifier.notifyOnCreate(EntryViews.readOnly(key, iv));
+         }
+
+         entry.setValue(value);
+         entry.setChanged(true);
+         updateEntryMetaParamsIfPresent(entry, metas);
+         return null;
+      }
+
+      @Override
+      public Void remove() {
+         entry.setRemoved(true);
+         entry.setChanged(true);
+         // For remove write-only listener events, create a value-less read entry view
+         //notifier.notifyOnWrite(EntryViews.noValue(entry.getKey()));
+         return null;
+      }
+
+      @Override
+      public <T> T getMetaParam(Class<T> type) {
+         Metadata metadata = entry.getMetadata();
+         if (metadata instanceof MetaParamsInternalMetadata) {
+            MetaParamsInternalMetadata metaParamsMetadata = (MetaParamsInternalMetadata) metadata;
+            return metaParamsMetadata.getMetaParam(type);
+         }
+
+         // TODO: Add interoperability support, e.g. able to retrieve lifespan for data stored in Cache via lifespan API
+
+         throw new NoSuchElementException("Metadata with type=" + type + " not found");
+      }
+
+      @Override
+      public <T> Optional<T> findMetaParam(Class<T> type) {
+         Metadata metadata = entry.getMetadata();
+         if (metadata instanceof MetaParamsInternalMetadata) {
+            MetaParamsInternalMetadata metaParamsMetadata = (MetaParamsInternalMetadata) metadata;
+            return metaParamsMetadata.findMetaParam(type);
+         }
+
+         // TODO: Add interoperability support, e.g. able to retrieve lifespan for data stored in Cache via lifespan API
```

Comment:
No, I meant that the existing Cache API provides a way to pass in lifespan, e.g. via `put(K, V, lifespan)`. The comment here is that there should be interoperability. If I stored something with lifespan via Cache, I should be able to locate that lifespan MetaParam via the functional API.